airbyte-integrations/connectors/source-snowflake/src/main/java/io.airbyte.integrations.source.snowflake/SnowflakeSourceOperations.java

```diff
@@ -86,11 +125,43 @@ public void setJsonField(final ResultSet resultSet, final int colIndex, final Ob
       case CHAR, VARCHAR, LONGVARCHAR -> putString(json, columnName, resultSet, colIndex);
       case DATE -> putDate(json, columnName, resultSet, colIndex);
       case TIME -> putTime(json, columnName, resultSet, colIndex);
-      case TIMESTAMP, TIMESTAMP_WITH_TIMEZONE -> putTimestamp(json, columnName, resultSet, colIndex);
+      case TIMESTAMP -> putTimestamp(json, columnName, resultSet, colIndex);
+      case TIMESTAMP_WITH_TIMEZONE -> putTimestampWithTimezone(json, columnName, resultSet, colIndex);
       case BLOB, BINARY, VARBINARY, LONGVARBINARY -> putBinary(json, columnName, resultSet, colIndex);
       case ARRAY -> putArray(json, columnName, resultSet, colIndex);
       default -> putDefault(json, columnName, resultSet, colIndex);
     }
   }
 
+  @Override
+  protected void setDate(final PreparedStatement preparedStatement, final int parameterIndex, final String value) throws SQLException {
+    final LocalDate date = LocalDate.parse(value);
+    preparedStatement.setDate(parameterIndex, Date.valueOf(date));
```

Comment:
does it cause problems to use setObject and pass in the localdate directly?
```java
preparedStatement.setObject(parameterIndex, LocalTime.parse(value));
```
Converting back to the java.sql.Date type makes me a little nervous

Reply:
> does it cause problems to use setObject and pass in the localdate directly?
> 
> ```java
> preparedStatement.setObject(parameterIndex, LocalTime.parse(value));
> ```
> 
> Converting back to the java.sql.Date type makes me a little nervous

yes, we cannot use the same way as it is implemented for Postgres. Snowflake preparedStatement is not friendly for classes like LocalDate, LocalTime, etc. if you try to execute the command `preparedStatement.setObject(parameterIndex, LocalDate.parse(value));`
 it will throw an exception
`net.snowflake.client.jdbc.SnowflakeSQLLoggedException: Data type not supported for binding: Object type: class java.time.LocalDate.`

Reply:
woooow

thanks for checking, didn't realize the snowflake jdbc driver works like that :/